registry_module.go

```diff
@@ -607,26 +634,46 @@ func (r *registryModules) Delete(ctx context.Context, organization, name string)
 }
 
 // DeleteProvider is used to delete the specific registry module provider
+// The implementation of this method has been replaced, see API docs:
+// https://developer.hashicorp.com/terraform/cloud-docs/api-docs/private-registry/modules#delete-a-module
+// See deprecatedDeleteProvider() for original implementation
 func (r *registryModules) DeleteProvider(ctx context.Context, moduleID RegistryModuleID) error {
-	if err := moduleID.valid(); err != nil {
+	if err := moduleID.validateOnProvider(); err != nil {
 		return err
 	}
 
+	var Namespace string
+
+	if moduleID.RegistryName == PrivateRegistry {
+		Namespace = moduleID.Organization
+	}
+
+	if moduleID.RegistryName == PublicRegistry {
+		Namespace = moduleID.Namespace
+	}
```

Comment:
Should It be incumbent upon the caller to provide this field within the moduleID struct? There is already a Namespace field there that we are ignoring.

To simplify all of this what if we added two methods for building a moduleID?

`func NewPublicRegistryModuleID(namespace, name, provider string) RegistryModuleID`

`func NewPrivateRegistryModuleID(organization, name, provider string) RegistryModuleID`

Perhaps that would simplify DeleteByName to allow it to use this struct as well?

This could apply to all three methods

Reply:
Also, local variables should always be lowercased although I think we should get rid of this one